k8s-multus/scripts/provision.sh

```diff
@@ -0,0 +1,158 @@
+yum -y remove NetworkManager firewalld
+
+# Required for iscsi demo to work.
+yum -y install iscsi-initiator-utils
+
+cat <<EOF >/etc/yum.repos.d/kubernetes.repo
+[kubernetes]
+name=Kubernetes
+baseurl=http://yum.kubernetes.io/repos/kubernetes-el7-x86_64
+enabled=1
+gpgcheck=1
+repo_gpgcheck=1
+gpgkey=https://packages.cloud.google.com/yum/doc/yum-key.gpg
+       https://packages.cloud.google.com/yum/doc/rpm-package-key.gpg
+EOF
+yum install -y docker
+
+# Log to json files instead of journald
+sed -i 's/--log-driver=journald //g' /etc/sysconfig/docker
+echo '{ "insecure-registries" : ["registry:5000"] }' > /etc/docker/daemon.json
+
+# Omit pgp checks until https://github.com/kubernetes/kubeadm/issues/643 is resolved.
+yum install --nogpgcheck -y \
+    kubeadm-${version} \
+    kubelet-${version} \
+    kubectl-${version} \
+    kubernetes-cni \
+    git \
+    golang
+
+# Latest docker on CentOS uses systemd for cgroup management
+cat <<EOT >>/etc/systemd/system/kubelet.service.d/09-kubeadm.conf
+[Service]
+Environment="KUBELET_EXTRA_ARGS=--cgroup-driver=systemd --runtime-cgroups=/systemd/system.slice --kubelet-cgroups=/systemd/system.slice"
+EOT
+systemctl daemon-reload
+
+systemctl enable docker && systemctl start docker
+systemctl enable kubelet && systemctl start kubelet
+
+# Needed for kubernetes service routing and dns
+# https://github.com/kubernetes/kubernetes/issues/33798#issuecomment-250962627
+modprobe bridge
+cat <<EOF >  /etc/sysctl.d/k8s.conf
+net.bridge.bridge-nf-call-ip6tables = 1
+net.bridge.bridge-nf-call-iptables = 1
+EOF
+sysctl --system
+
+kubeadm init --pod-network-cidr=10.244.0.0/16 --kubernetes-version v${version} --token abcdef.1234567890123456
+
+# install multus
+git clone https://github.com/intel/multus-cni.git -b dev/network-plumbing-working-group-crd-change
+cd multus-cni
+./build
+
+cp bin/multus /opt/cni/bin/
+
+cd examples
+
+cat > macvlan-conf.yml <<EOF
+apiVersion: "k8s.cni.cncf.io/v1"
+kind: NetworkAttachmentDefinition
+metadata:
+  name: macvlan-conf
+spec: 
+  config: '{
+      "cniVersion": "0.3.0",
+      "type": "macvlan",
+      "master": "eth0",
+      "mode": "bridge",
+      "ipam": {
+        "type": "host-local",
+        "subnet": "192.168.66.0/24",
+        "rangeStart": "192.168.66.200",
+        "rangeEnd": "192.168.66.216",
+        "routes": [
+          { "dst": "0.0.0.0/0" }
+        ],
+        "gateway": "192.168.66.2"
+      }
+    }'
+EOF
+
+kubectl --kubeconfig=/etc/kubernetes/admin.conf create -f clusterrole.yml
+kubectl --kubeconfig=/etc/kubernetes/admin.conf create clusterrolebinding multus-node-`hostname` \
+                                                       --clusterrole=multus-crd-overpowered \
+                                                       --user=system:node:`hostname`
+
+kubectl --kubeconfig=/etc/kubernetes/admin.conf apply -f https://raw.githubusercontent.com/intel/multus-cni/dev/network-plumbing-working-group-crd-change/examples/multus-with-flannel.yml
```

Comment:
Can you download the file here and then just provision it from the file here and in the startup script? You don't have to repeate a mistake which we are going to fix: #23.